core/src/main/scala/pureconfig/syntax/package.scala

```diff
@@ -31,4 +32,7 @@ package object syntax {
     def to[T](implicit reader: Derivation[ConfigReader[T]]): Either[ConfigReaderFailures, T] = conf.root().to[T]
     def toOrThrow[T](implicit reader: Derivation[ConfigReader[T]], cl: ClassTag[T]): T = getResultOrThrow(conf.root().to[T])(cl)
   }
+
+  implicit def intToPathSegment(i: Int): Path.Segment = Path.Index(i)
```

Comment:
These conversions should be better inside the companion object of `Path` or `Path.Segment`. They do not need to be imported explicitly if they're there.